src/core/server/saved_objects/migrationsv2/actions/index.ts

```diff
@@ -322,6 +343,20 @@ const waitForTask = (
         description: body.task.description,
       });
     })
+    .catch((e) => {
```

Comment:
Let's extract it in a function as you did for `catchRetryableEsClientErrors`? I like the fact it has an explicit signature.